gateway/ln-gateway/src/db.rs

```diff
@@ -341,16 +373,25 @@ pub struct GatewayConfigurationV1 {
 }
 
 #[derive(Debug, Clone, Eq, PartialEq, Encodable, Decodable)]
-pub struct GatewayConfigurationKey;
+pub struct GatewayConfigurationKeyV2;
 
 #[derive(Debug, Clone, Eq, PartialEq, Encodable, Decodable, Serialize, Deserialize)]
-pub struct GatewayConfiguration {
+pub struct GatewayConfigurationV2 {
     pub num_route_hints: u32,
     #[serde(with = "serde_routing_fees")]
     pub routing_fees: RoutingFees,
     pub network: NetworkLegacyEncodingWrapper,
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, Encodable, Decodable)]
+pub struct GatewayConfigurationKey;
+
+#[derive(Debug, Clone, Eq, PartialEq, Encodable, Decodable, Serialize, Deserialize)]
+pub struct GatewayConfiguration {
+    pub num_route_hints: u32,
+    pub network: NetworkLegacyEncodingWrapper,
```

Comment:
Please migrate to the new network encoding while you are at it.

Reply:
Maybe we should just remove it altogether and make it an environment variable? It doesn't make sense to me to put this in the `GatewayConfiguration`, since that implies that you'd want to change it at runtime without restarting. To me, changing networks seems fine to require a restart.

Bonus is that `num_route_hints` would be the only thing left here and that will go away once LNv1 is gone

Reply:
This is removed entirely